_posts/2020-02-14-diy-drones-digital.md

```diff
@@ -0,0 +1,171 @@
+    - The ESC does one of three things with the PWM
+        - **Goes into startup:** This is sort of a motor priming step, any changes in the signal from here on will result in the motor spinning up. This is usually at the lowest throttle position during startup, 1ms PWM.
+        - **Actually spins up the motors:** This is post priming, the ESC drives the motor depending upon the PWM input it receives. Ranges from 1ms being 0% RPM to 2ms being 100% RPM.
+        - **Setup Mode:** This is usually to enter a mode where you can sync the transmitter's min and max PWM signals to the ESC to make sure that the throttle response is accurate. This is the highest throttle position during startup, 2ms PWM.
+        - **Makes Beeping noise:** ESCs vibrate the motors in order to make beeping noise, this is the ONLY output of the ESC.
+
+***
+
+### The Actual Flying bit:
+- ESC calibrate: (First flight only)
+    - The controller is set to its highest position.
+    - The receiver is switched on.
+    - An individual ESC is switched on, till it beeps a tune to indicate that it is in calibration mode.
+    - Move the throttle to its lowest position and wait for the beep.
+    - Move the throttle to its highest position and wait for the beep.
+    - Power off the esc.
+    - Repeat the steps for all the other ESCs.
+- Flying:
+    - Turn on the transmitter and the receiver.
+    - Set the throttle to the lowest point.
+    - Turn on an individual ESC, wait for the "Primed" tone.
+    - Repeat for all ESCs one at a time.
+    - At this point you can start flying the drone.
+
+So As you just read, even though the ESC itself is a 32bit Micro, the way it functions and the limitations it possess are all affected by the fact that every thing is controlled over a single PWM signal.
+
+***
+***
+
+# The Problems and Limitations.
+
+### Is PWM Communication Analog or Digital?
+In my opinion, PWM Communication is an analog interface that is meant to be used with digital logic. It is the closest analog signal that can be natively generated by a digital system without using auxiliary circuitry like a ADC and DAC.
+
+**Why is it not digital?**
+It doesn't transmit data in bits... 0s & 1s. The transmissions relies on the amount of time the signal is held high during a cycle. in a 2ms wide cycle, 0ms is 0% pwm, 1ms is 50% pwm and 2ms is 100% pwm... and everything in between.
+Whereas in a digital communication standard, the data is transmitted in bits and each word can be represented as a set decimal or hex value.
+
+**Why is it closer to analogue?**
+Sample rate doesn't matter... as much.
+If the sample rate of the transmitter is 100HZ and the receiver is polling at 1KHZ, it can still interpret the PWM signal perfectly. Its is a bit less lenient if the receiver is polling at a lower frequency, but as long as its not too far off, say 90Hz in this example. The receiver can still interpret accurately.
```

Comment:
```suggestion
If the sample rate of the transmitter is 100HZ and the receiver is polling at 1KHZ, it can still interpret the PWM signal perfectly. It is a bit less lenient if the receiver is polling at a lower frequency, but as long as it's not too far off, say 90Hz in this example. The receiver can still interpret accurately.
```